pkg/publish.go

```diff
@@ -59,15 +60,78 @@ func PrePublish(file string, rootID string, dryRun bool) (id string, err error)
 		),
 	)
 	if err != nil {
-		return
+		return err
+	}
+
```

Comment:
Seems like we should explicitly create a ROOT source here, before looping over all the files.

Reply:
This is essentially what happens. First, find all eligible source files. Out of those source files, see if one is the root page. If one exists, use it, or use the default generated page. Stub the root for the ID, then publish all files.

There is some cheating going on, since we (harmlessly) stub the root page twice. I think I prefer this way, however, as if a page dependency management system is implemented, I think it would behave in a similar way. Find all files, then publish them in an appropriate order (root page likely first). The root page being published first is hard coded here instead.